CODE/Sources/PennyBOT/Handlers/InteractionHandler.swift

```diff
@@ -1,104 +1,182 @@
+                    .get(discordID: discordId)
+                    .enumerated().map { idx, text in
+                        "**\(idx).** `\(text)`"
+                    }.joined(separator: "\n")
+                return list
+            default:
+                logger.error("Unrecognized link option", metadata: ["name": "\(first.name)"])
+                return "Option not recognized: \(first.name)"
             }
         } catch {
-            logger.error("Discord Client error", metadata: ["error": "\(error)"])
-            return false
+            logger.error("Pings command error", metadata: ["error": "\(error)"])
+            return "Sorry some errors happened :( please report this to us if it happens again"
         }
     }
     
-    private func respond(with response: String) async {
-        do {
-            let apiResponse = try await discordClient.editInteractionResponse(
-                token: event.token,
-                payload: .init(
-                    embeds: [.init(
-                        description: response,
-                        color: .vaporPurple
-                    )]
-                )
+    /// Returns `true` if the acknowledgement was successfully sent
+    private func sendInteractionAcknowledgement(isEphemeral: Bool) async -> Bool {
+        await DiscordService.shared.respondToInteraction(
+            id: event.id,
+            token: event.token,
+            payload: .init(
+                type: .deferredChannelMessageWithSource,
+                data: isEphemeral ? .init(flags: [.ephemeral]) : nil
             )
-            if !(200..<300).contains(apiResponse.status.code) {
-                logger.report(
-                    "Received non-200 status from Discord API for interaction",
-                    response: apiResponse
-                )
-            }
-        } catch {
-            logger.error("Discord Client error", metadata: ["error": "\(error)"])
+        )
+    }
+    
+    private func sendInteractionNameResolveFailure() async {
+        await DiscordService.shared.respondToInteraction(
+            id: event.id,
+            token: event.token,
+            payload: .init(
+                type: .channelMessageWithSource,
+                data: .init(embeds: [.init(
+                    description: "Failed to resolve the interaction name :(",
+                    color: .vaporPurple
+                )], flags: [.ephemeral])
+            )
+        )
+    }
+    
+    private func respond(with response: String) async {
+        await DiscordService.shared.editInteraction(
+            token: event.token,
+            payload: .init(embeds: [.init(
+                description: response,
+                color: .vaporPurple
+            )])
+        )
+    }
+}
+
+private enum SlashCommandKind {
+    case link
+    case automatedPings
+    
+    /// Ephemeral means the interaction will only be visible to the user, not the whole guild.
+    var isEphemeral: Bool {
```

Comment:
Both `link` and `auto-pings` use `ephemeral` interaction messages. Meaning that the interaction-message is not visible to other people at all. Only the user of the command sees the messages.

for auto-pings this is important to make sure people don't abuse the bot by seeing what one person is registering as a word to get pinged for, then intentionally spamming that word just to ping the person.